js/apps/admin-ui/src/clients/authorization/ResourcesPolicySelect.tsx

```diff
@@ -108,57 +128,133 @@ export const ResourcesPolicySelect = ({
     [search]
   );
 
+  const [toggleUnsavedChangesDialog, UnsavedChangesConfirm] = useConfirmDialog({
+    titleKey: t("common:unsavedChangesTitle"),
+    messageKey: t("common:unsavedChangesConfirm"),
+    continueButtonLabel: t("common:continue"),
+    continueButtonVariant: ButtonVariant.danger,
+    onConfirm: async () => {
+      navigate(
+        toPolicyDetails({
+          realm: realm,
+          id: clientId,
+          policyId: clickedPolicy?.id!,
+          policyType: clickedPolicy?.type!,
+        })
+      );
+    },
+  });
+
   const toSelectOptions = () =>
     items.map((p) => (
       <SelectOption key={p.id} value={p.id}>
         {p.name}
       </SelectOption>
     ));
 
+  const toChipGroupItems = (
+    field: ControllerRenderProps<PolicyRepresentation, Type>
+  ) => {
+    return (
+      <ChipGroup>
+        {(field.value as string[]).map((permissionId) => {
+          const policy = items.find(
+            (permission) => permission.id === permissionId
+          );
+          if (policy) {
```

Comment:
I am more of a fan of inverting this condition and using a early return. This prevents deeply nested statements, and improves readability of the code.

```ts
if (!policy) {
  return null;
}

// The rest of the code.
```